### PR TITLE
docs: Extend memory limit for Workload with large SBOM

### PR DESCRIPTION
**Pull request by rashedkvm**
# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?
1-7-0, main
It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

### CODE REVIEW

**Comment by RichardJJG** [comment]
We received your PR after the Doc Freeze date. We are currently working on release tasks, and we will review your PR as soon as we are able. Unfortunately, this might mean that your PR content will not be published until after the release. Going forward, to ensure that your PR changes are published on release day, please submit such PRs no later than 23:59 UTC-8 on the [Doc Freeze date](https://confluence.eng.vmware.com/pages/viewpage.action?pageId=789919969).

File: cartographer-conventions/troubleshooting.hbs.md [hunk 1]
```diff
@@ -183,3 +183,87 @@ imagePullSecrets:
 secrets:
 - name: registry-credentials
 ``` 
+
+## <a id="oom-killed"></a> `OOMKilled` Convention Controller 
+
+### Symptoms
+
+While processing workloads with large SBOM, the cartographer convention controller manager pod can 
+be failing with status `CrashLoopBackOff` or `OOMKilled`. For example:
```
Comment (rszumlakowski) [inline cartographer-conventions/troubleshooting.hbs.md:192]:
```suggestion
While processing workloads with large SBOM, the Cartographer Convention controller manager pod can 
fail with status the `CrashLoopBackOff` or `OOMKilled`. For example:
```

File: cartographer-conventions/troubleshooting.hbs.md [hunk 2]
```diff
@@ -183,3 +183,87 @@ imagePullSecrets:
...
+
+```console
+NAME                                                          READY   STATUS             RESTARTS          AGE
+cartographer-controller-6996774647-bs98l                      1/1     Running            0                 6d7h
+cartographer-conventions-controller-manager-ff4cdf59d-5nzl5   0/1     CrashLoopBackOff   1292 (109s ago)   5d3h
+```
+
+Controller pod staltus may look like the following.
```
Comment (rszumlakowski) [inline cartographer-conventions/troubleshooting.hbs.md:200]:
```suggestion
The controller pod status may look like the following:
```

File: cartographer-conventions/troubleshooting.hbs.md [hunk 3]
```diff
@@ -183,3 +183,87 @@ imagePullSecrets:
...
+        reason: OOMKilled
+        startedAt: "2023-11-06T21:02:10Z"
+    name: manager
+```
+
+### Cause
+
+This error usually occurs when a `workload` image, built by the supply chain, contains a large SBOM.
+In some cases, the default resource limit set during installation may not be enough to process the 
+pod conventions which can lead to a crashing pod.
```
Comment (rszumlakowski) [inline cartographer-conventions/troubleshooting.hbs.md:221]:
```suggestion
This error usually occurs when a `workload` image, built by the supply chain, contains a large SBOM.
In some cases, the default resource limit set during installation may not be enough to process the 
pod conventions which can lead to the controller pod crashing.  
```

File: cartographer-conventions/troubleshooting.hbs.md [hunk 4]
```diff
@@ -183,3 +183,87 @@ imagePullSecrets:
...
+### Cause
+
+This error usually occurs when a `workload` image, built by the supply chain, contains a large SBOM.
+In some cases, the default resource limit set during installation may not be enough to process the 
+pod conventions which can lead to a crashing pod.  
+
+### Solution 
+
+Using a ytt overlay, cartographer convention controller manager memory limit can be increased post 
+TAP installation. Here is an exmaple of a ytt overlay to icrease the memory limit.
```
Comment (rszumlakowski) [inline cartographer-conventions/troubleshooting.hbs.md:226]:
```suggestion
By using a ytt overlay, the Cartographer Convention controller manager memory limit can be increased after 
TAP installation. Here is an exmaple of a ytt overlay to increase the memory limit.
```

File: cartographer-conventions/troubleshooting.hbs.md [hunk 5]
```diff
@@ -183,3 +183,87 @@ imagePullSecrets:
...
+
+1. Create a Secret with the following ytt overlay.
```
Comment (rszumlakowski) [inline cartographer-conventions/troubleshooting.hbs.md:228]:
```suggestion
1. Create a Secret with the following ytt overlay:
```

File: cartographer-conventions/troubleshooting.hbs.md [hunk 6]
```diff
@@ -183,3 +183,87 @@ imagePullSecrets:
...
+          containers:
+            #@overlay/match by=overlay.subset({"name": "manager"})
+            - name: manager
+              resources:
+                limits:
+                  cpu: 100m
+                  memory: 512Mi
+```
+
+2. Update the TAP values YAML file to include a package_overlays field:
```
Comment (rszumlakowski) [inline cartographer-conventions/troubleshooting.hbs.md:254]:
```suggestion
2. Update the TAP values YAML file to include a `package_overlays` field:
```

File: cartographer-conventions/troubleshooting.hbs.md [hunk 7]
```diff
@@ -183,3 +183,87 @@ imagePullSecrets:
...
+
+```yaml
+package_overlays:
+- name: Cartographer
```
Comment (rszumlakowski) [inline cartographer-conventions/troubleshooting.hbs.md:258]:
```suggestion
- name: cartographer
```

File: release-notes.hbs.md [hunk 8]
```diff
@@ -1272,6 +1272,108 @@ This release has the following known issues, listed by component and area.
   and restrict access to all or parts of Tanzu Developer Portal.
   For more information, see [Troubleshooting](tap-gui/troubleshooting.hbs.md#ad-block-interference).
 
+#### <a id='1-7-0-convention-ki'></a> v1.7.0 Known issues: Convention OOMKilled
+
+While processing workloads with large SBOM, the cartographer convention controller manager pod can be failing with status `CrashLoopBackOff` or `OOMKilled`.
```
Comment (rszumlakowski) [inline release-notes.hbs.md:1277]:
```suggestion
While processing workloads with large SBOMs, the Cartographer Convention controller manager pod can fail with  the status `CrashLoopBackOff` or `OOMKilled`. 
```

File: release-notes.hbs.md [hunk 9]
```diff
@@ -1272,6 +1272,108 @@ This release has the following known issues, listed by component and area.
   and restrict access to all or parts of Tanzu Developer Portal.
   For more information, see [Troubleshooting](tap-gui/troubleshooting.hbs.md#ad-block-interference).
 
+#### <a id='1-7-0-convention-ki'></a> v1.7.0 Known issues: Convention OOMKilled
+
+While processing workloads with large SBOM, the cartographer convention controller manager pod can be failing with status `CrashLoopBackOff` or `OOMKilled`. 
+
+For a workaround, increasing the memory limit to `512Mi` has proven to fix the pod crash. Please see the steps to increase the memory limit in [Cartographer Convention documentation](/cartographer-conventions/troubleshooting.hbs.md#oomkilled-convention-controller)
```
Comment (rszumlakowski) [inline release-notes.hbs.md:1279]:
```suggestion
To work around this problem you can increase the memory limit to `512Mi`. Please see the steps to increase the memory limit in [Cartographer Convention documentation](/cartographer-conventions/troubleshooting.hbs.md#oomkilled-convention-controller).
```

File: release-notes.hbs.md [hunk 10]
```diff
@@ -1272,6 +1272,108 @@ This release has the following known issues, listed by component and area.
...
+
+Similar memory limit may need to be increased for the convention webhook servers. Use the following steps to increase the memory limit for the following convention servers,
```
Comment (rszumlakowski) [inline release-notes.hbs.md:1281]:
```suggestion
You may need to increase the memory limit for the various Convention webhook servers. Use the following steps to increase the memory limit for the following Convention servers:
```

File: release-notes.hbs.md [hunk 11]
```diff
@@ -1272,6 +1272,108 @@ This release has the following known issues, listed by component and area.
...
+- app-live-view-conventions
+- spring-boot-webhook
+- developer-conventions/webhook
+
+1. Create a Secret with the following ytt overlay.
```
Comment (rszumlakowski) [inline release-notes.hbs.md:1286]:
```suggestion
1. Create a `Secret` with the following ytt overlay:
```

File: cartographer-conventions/troubleshooting.hbs.md (hunk 1 above)
Comment (rashedkvm) [reply cartographer-conventions/troubleshooting.hbs.md:192]: you meant `with the status` :)

**Pull request merged**